include/rviz_visual_tools/remote_control.hpp

```diff
@@ -157,10 +145,10 @@ class RemoteControl
   // Input
   rclcpp::Subscription<sensor_msgs::msg::Joy>::SharedPtr rviz_dashboard_sub_;
 
-  std::unique_ptr<std::promise<void>> next_step_ready_;
+  bool is_waiting_ = false;
+  bool next_step_ready_ = false;
```

Comment:
I believe you should add a `mutex` to protect the access to more than 1 variable.